File: solutions/haskell/etl/1/src/ETL.hs
module ETL (transform) where

import Data.Char (toLower)
import Data.Map (Map)
import qualified Data.Map as M

transform :: Map a String -> Map Char a
transform = M.foldlWithKey' f M.empty
  where
    f a k v = let ins e = M.insert (toLower e) k
              in foldr ins a v

File: solutions/go/etl/1/etl.go
package etl

func Transform(in map[int][]string) map[string]int {
	out := make(map[string]int)
    for k, v := range in {
        for _, s := range v {
            out[string([]rune(s)[0] - 'A' + 'a')] = k
        }
    }
    return out
}
